frontend/src/components/DateRangeSelect.js

```diff
@@ -256,17 +260,32 @@ function DateRangeSelect(props) {
                     focused={startDateFocused}
                     numberOfMonths={1}
                     hideKeyboardShortcutsPanel
-                    isOutsideRange={(day) => isOutsideRange(day, 'start')}
                     onFocusChange={({ focused }) => {
                       if (!focused) {
                         setStartDateFocused(focused);
                       }
                     }}
+                    isOutsideRange={(day) => isOutsideRange(day, 'start')}
                     onDateChange={(selectedDate) => {
-                      setStartDate(selectedDate);
+                      // weird that we'd have to explicitly do this
+                      // but otherwise things get all wacky when you
+                      // type in a date that's different than the end date
+                      if (!selectedDate) {
```

Comment:
Just a guess, `onDateChange` is called for every character that is typed and selectedDate is null when the data is invalid?

Reply:
you are correct sir